src/UI/Components/Diagram/Canvas.tsx

```diff
@@ -54,6 +55,21 @@ const Canvas = ({
     service: mainServiceName,
   });
 
+  const handleLooseEmbeddedEvent = (event) => {
+    const customEvent = event as CustomEvent;
```

Comment:
can't you type the argument instead?

Reply:
unfortunately AddEventListener accept function where its argument it of type Event, which is missing property "detail" that stores, tried to override type in the listener itself but it throws out error, even though CustomEvent extends Event interface

Reply:
right... forgot Typescript was sometimes complicated.